[2025-04-04T22:07:41.203+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-04T22:07:41.425+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: scrape_amazon_dag.process_Tools___Equipment_12 manual__2025-04-04T22:06:59.454233+00:00 [queued]>
[2025-04-04T22:07:41.463+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: scrape_amazon_dag.process_Tools___Equipment_12 manual__2025-04-04T22:06:59.454233+00:00 [queued]>
[2025-04-04T22:07:41.511+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2025-04-04T22:07:41.661+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): process_Tools___Equipment_12> on 2025-04-04 22:06:59.454233+00:00
[2025-04-04T22:07:41.702+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=2198) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-04-04T22:07:41.718+0000] {standard_task_runner.py:72} INFO - Started process 2252 to run task
[2025-04-04T22:07:41.736+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'scrape_amazon_dag', 'process_Tools___Equipment_12', 'manual__2025-04-04T22:06:59.454233+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/test_crawl_list_product.py', '--cfg-path', '/tmp/tmp7zug3cm1']
[2025-04-04T22:07:41.760+0000] {standard_task_runner.py:105} INFO - Job 10: Subtask process_Tools___Equipment_12
[2025-04-04T22:07:49.011+0000] {task_command.py:467} INFO - Running <TaskInstance: scrape_amazon_dag.process_Tools___Equipment_12 manual__2025-04-04T22:06:59.454233+00:00 [running]> on host ba893f484bac
[2025-04-04T22:07:51.757+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='scrape_amazon_dag' AIRFLOW_CTX_TASK_ID='process_Tools___Equipment_12' AIRFLOW_CTX_EXECUTION_DATE='2025-04-04T22:06:59.454233+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-04T22:06:59.454233+00:00'
[2025-04-04T22:07:51.777+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-04T22:07:51.783+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-04T22:07:51.801+0000] {logging_mixin.py:190} INFO - Current task name:process_Tools___Equipment_12 state:running start_date:2025-04-04 22:07:41.431997+00:00
[2025-04-04T22:07:51.805+0000] {logging_mixin.py:190} INFO - Dag name:scrape_amazon_dag and current dag run status:running
[2025-04-04T22:07:51.806+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-04-04T22:07:51.851+0000] {test_crawl_list_product.py:227} INFO - Processing: Automotive/Tools & Equipment - https://www.amazon.com/s?bbn=2562090011&rh=i%3Aspecialty-aps%2Cn%3A2562090011%2Cn%3A%2115690151%2Cn%3A15706941&s=date-desc-rank&ref_=nav_em__nav_desktop_sa_intl_tools_equipment_0_2_10_14
[2025-04-04T22:07:52.117+0000] {logger.py:11} INFO - ====== WebDriver manager ======
[2025-04-04T22:07:52.638+0000] {logger.py:11} INFO - Get LATEST chromedriver version for google-chrome
[2025-04-04T22:07:53.508+0000] {logger.py:11} INFO - About to download new driver from https://chromedriver.storage.googleapis.com/114.0.5735.90/chromedriver_linux64.zip
[2025-04-04T22:07:53.617+0000] {logger.py:11} INFO - Driver downloading response is 200
[2025-04-04T22:08:10.361+0000] {logger.py:11} INFO - Get LATEST chromedriver version for google-chrome
[2025-04-04T22:08:11.794+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/test_crawl_list_product.py", line 228, in scrape_department
    scraper = AmazonScraper()
              ^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/test_crawl_list_product.py", line 41, in __init__
    self.setup_driver()
  File "/opt/airflow/dags/test_crawl_list_product.py", line 56, in setup_driver
    service = ChromeService(ChromeDriverManager().install())
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/chrome.py", line 40, in install
    driver_path = self._get_driver_binary_path(self.driver)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/manager.py", line 41, in _get_driver_binary_path
    binary_path = self._cache_manager.save_file_to_cache(driver, file)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/driver_cache.py", line 54, in save_file_to_cache
    files = self.unpack_archive(archive, path)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/driver_cache.py", line 49, in unpack_archive
    return self._file_manager.unpack_archive(archive, path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/file_manager.py", line 57, in unpack_archive
    return self.__extract_zip(archive_file, target_dir)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/file_manager.py", line 63, in __extract_zip
    archive = zip_class(archive_file.file_path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/zipfile/__init__.py", line 1349, in __init__
    self._RealGetContents()
  File "/usr/local/lib/python3.12/zipfile/__init__.py", line 1416, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-04-04T22:08:11.921+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-04T22:08:11.953+0000] {logging_mixin.py:190} INFO - Task start:2025-04-04 22:07:41.431997+00:00 end:2025-04-04 22:08:11.920802+00:00 duration:30.488805
[2025-04-04T22:08:12.001+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): process_Tools___Equipment_12> dag:<DAG: scrape_amazon_dag> dagrun:<DagRun scrape_amazon_dag @ 2025-04-04 22:06:59.454233+00:00: manual__2025-04-04T22:06:59.454233+00:00, state:running, queued_at: 2025-04-04 22:06:59.580718+00:00. externally triggered: True>
[2025-04-04T22:08:12.003+0000] {logging_mixin.py:190} INFO - Failure caused by File is not a zip file
[2025-04-04T22:08:12.096+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=scrape_amazon_dag, task_id=process_Tools___Equipment_12, run_id=manual__2025-04-04T22:06:59.454233+00:00, execution_date=20250404T220659, start_date=20250404T220741, end_date=20250404T220811
[2025-04-04T22:08:12.368+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-04-04T22:08:12.369+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 10 for task process_Tools___Equipment_12 (File is not a zip file; 2252)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/test_crawl_list_product.py", line 228, in scrape_department
    scraper = AmazonScraper()
              ^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/test_crawl_list_product.py", line 41, in __init__
    self.setup_driver()
  File "/opt/airflow/dags/test_crawl_list_product.py", line 56, in setup_driver
    service = ChromeService(ChromeDriverManager().install())
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/chrome.py", line 40, in install
    driver_path = self._get_driver_binary_path(self.driver)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/manager.py", line 41, in _get_driver_binary_path
    binary_path = self._cache_manager.save_file_to_cache(driver, file)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/driver_cache.py", line 54, in save_file_to_cache
    files = self.unpack_archive(archive, path)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/driver_cache.py", line 49, in unpack_archive
    return self._file_manager.unpack_archive(archive, path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/file_manager.py", line 57, in unpack_archive
    return self.__extract_zip(archive_file, target_dir)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/webdriver_manager/core/file_manager.py", line 63, in __extract_zip
    archive = zip_class(archive_file.file_path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/zipfile/__init__.py", line 1349, in __init__
    self._RealGetContents()
  File "/usr/local/lib/python3.12/zipfile/__init__.py", line 1416, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
[2025-04-04T22:08:12.473+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-04-04T22:08:12.516+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
